plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -1283,12 +1283,15 @@ private static Location getValidatedExternalLocation(String location)
         }
     }
 
-    private void checkExternalPath(HdfsContext context, Path path)
+    private void checkExternalPathAndCreateIfNotExists(HdfsContext context, Path path)
     {
         try {
             if (!isS3FileSystem(context, hdfsEnvironment, path)) {
```

Comment:
As far as I understand on s3 we we don't have the same abstraction on directories as on hdfs.
Directories will be just part of the key, when we create file and will be created automatically when we create file.

Reply:
So we could skip the check right ? Since creating a directory in S3 would be a no-op.

Reply:
Yes, but then it will be call to 
`if (!fileSystem.isDirectory(path)) `

and this will fail for s3

Reply:
> Yes, but then it will be call to `if (!fileSystem.isDirectory(path)) `
> 
> and this will fail for s3

wrong, thx!!
anyway we need to check that it's not a file

Reply:
removed